custom_components/localtuya/common.py

```diff
@@ -89,13 +89,21 @@ def __get_status(self):
                     raise ConnectionError("Failed to update status .")
 
     def set_dps(self, state, dps_index):
-        # _LOGGER.info("running def set_dps from cover")
-        """Change the Tuya switch status and clear the cache."""
-        self._cached_status = ""
-        self._cached_status_time = 0
+        """Change value of a DP of the Tuya device and update the cached status."""
+        # _LOGGER.info("running def set_dps from TuyaDevice")
+        # No need to clear the cache here: let's just update the status of the
+        # changed dps as returned by the interface (see 5 lines below)
+        # self._cached_status = ""
+        # self._cached_status_time = 0
         for i in range(5):
             try:
-                return self._interface.set_dps(state, dps_index)
+                result = self._interface.set_dps(state, dps_index)
+                self._cached_status["dps"].update(result["dps"])
+                # NOW WE SHOULD TRIGGER status_updated FOR ALL ENTITIES
+                # INVOLVED IN result["dps"] :
+                # for dp in result["dps"]:
+                #    have status_updated() called....
```

Comment:
I think you should do something like this:

```python
from homeassistant.helpers.dispatcher import async_dispatcher_send
...
signal = f"localtuya_{self._interface.id}"
async_dispatcher_send(hass, signal, self._cached_status["status"])
```

Reply:
Yeah I thought of this, but the fact is that we don't have access to the `hass` object here... or am I wrong??

Reply:
Nah, not right now. You should pass it to the constructor and save it.

Reply:
Yeah, that's what I was thinking. I'll try this...

Reply:
OK it seems to be working now!
I'll push this PR, then we'll wait for #58 before merging everything.